src/Microsoft.DotNet.Build.Tasks/PackageRestoreLogger.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    public class PackageRestoreLogger : ILogger
+    {
+        public LoggerVerbosity Verbosity { get; set; }
+        public string Parameters { get; set; }
+
+        private HashSet<string> _packageSet = new HashSet<string>();
+        private string _logFile;
+        private bool _outputToConsole = true;
+        private Func<string, bool> _packageUrlCheck = x => x.Contains("OK") || x.Contains("CACHE");
+        private Func<string, bool> _packageCheck = x => x.Contains("nupkg");
+        private const string urlRegex = @"https:\/\/(?<domain>.*?(\.com|\.net|\.org)).*\.nupkg";
+
+        public void Initialize(IEventSource eventSource)
+        {
+            ParseParameters();
+            AttachToRequiredEvents(eventSource);
+        }
+
+        private void ParseParameters()
+        {
+            if (!String.IsNullOrEmpty(Parameters))
+            {
+                string[] parameterPairs = Parameters.Split(';');
+                foreach (string parameterPair in parameterPairs)
+                {
+                    if (parameterPair.Length > 0)
+                    {
+                        string[] parameterNameValue = parameterPair.Split('=');
+
+                        ApplyParameter(parameterNameValue[0], parameterNameValue.Length > 1 ? parameterNameValue[1] : null);
+                    }
+                }
+            }
+        }
+
+        private void ApplyParameter(string name, string value)
+        {
+            switch (name.ToLower())
+            {
+                case "logfile":
+                    _logFile = value;
+                    break;
+                case "logtoconsole":
+                    _outputToConsole = Boolean.Parse(value);
+                    break;
+                default:
+                    // ignore unrecognized parameters
+                    break;
+            }
+        }
+
+        private void AttachToRequiredEvents(IEventSource eventSource)
+        {
+            eventSource.MessageRaised += eventSource_MessageHandler;
+        }
+
+        private void eventSource_MessageHandler(object sender, BuildMessageEventArgs e)
+        {
+            if (_packageUrlCheck(e.Message) && _packageCheck(e.Message))
```

Comment:
Not certain that both of these comparisons are necessary.  I think you can just get away with "if X contains "OK", add it to the hashset.  doesn't matter if it's a nupkg (not sure what else would get restored), or internet vs local restore

Reply:
I've noticed that the OK is the status code response from the previous GET request. So in a restore we would see a set of requests to the index.json as well, which I was trying to exclude.
Sample:
```
OK https://api.nuget.org/v3-flatcontainer/microsoft.netcore.app/index.json 302ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:48 PM Message :                        OK https://api.nuget.org/v3-flatcontainer/microsoft.netcore.app/2.0.0/microsoft.netcore.app.2.0.0.nupkg 4ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:48 PM Message :                        OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.netcore.app/index.json 133ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:48 PM Message :                        OK https://dotnetfeed.blob.core.windows.net/dotnet-core/packages/microsoft.netcore.app/index.json 153ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:48 PM Message :                        OK https://dotnetfeed.blob.core.windows.net/dotnet-core/packages/microsoft.netcore.platforms/index.json 35ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:48 PM Message :                        OK https://dotnetmyget.blob.core.windows.net/artifacts/dotnet-core/nuget/v3/flatcontainer/microsoft.netcore.platforms/index.json 50ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:48 PM Message :                        OK https://dotnetfeed.blob.core.windows.net/dotnet-core/packages/microsoft.netcore.dotnethostpolicy/index.json 54ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:49 PM Message :                        OK https://dotnetfeed.blob.core.windows.net/dotnet-core/packages/netstandard.library/index.json 63ms (TaskId:30) 
RestoreTrackingLogger - 10/25/2017 5:29:49 PM Message :                        OK https://dotnetfeed.blob.core.windows.net/dotnet-core/packages/microsoft.netcore.dotnethostpolicy/2.0.0/microsoft.netcore.dotnethostpolicy.2.0.0.nupkg 
```

Reply:
I think it's ok to include it.

Reply:
Actually this might help us identify missing packages if the index.json at the time received an OK